include/Edge.hpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the JKQ DD Package which is released under the MIT license.
+ * See file README.md or go to http://iic.jku.at/eda/research/quantum_dd/ for more information.
+ */
+
+#ifndef DD_PACKAGE_EDGE_HPP
+#define DD_PACKAGE_EDGE_HPP
+
+#include "Complex.hpp"
+#include "ComplexTable.hpp"
+#include "ComplexValue.hpp"
+
+#include <cstddef>
+#include <utility>
+
+namespace dd {
+    template<class Node>
+    struct Edge {
+        Node*   p;
+        Complex w;
+
+        constexpr bool operator==(const Edge& other) const {
+            return p == other.p && w.approximatelyEquals(other.w);
```

Comment:
I think its rather surprising that `==` checks the equivalence approximately. Maybe we can discuss this in the next meeting.

Reply:
Ironically, `approximatelyEquals` was previously called `equals`. 
We should definitely discuss this!